### PR TITLE
dialects: (onnx) add onnx.EntryPoint

### LINKED ISSUE #2067: add onnx.EntryPoint

### PR DESCRIPTION
**Pull request by kayode-gif**
resolves #2067

### CODE REVIEW

**Comment by kayode-gif** [comment]
note btw: i didn't include any verification function, as i didn't believe there was anything further to verify (in the ```onnx-mlir``` there is not anything too verify also)

**Review by compor — APPROVED** [review]

File: tests/filecheck/dialects/onnx/onnx_ops.mlir [hunk 1]
```diff
@@ -67,3 +67,8 @@
 
 %res_max_pool_single_out = "onnx.MaxPoolSingleOut"(%t26) {onnx_node_name = "/MaxPoolSingleOut", "auto_pad" = "VALID", "ceil_mode" = 0 : i64, "kernel_shape" = [3 : i64, 3 : i64], "dilations" = [1 : i64, 1 : i64], "pads" = [0 : i64, 0 : i64, 0 : i64, 0 : i64], "storage_order" = 0 : i64, "strides" = [1 : i64, 1 : i64]}: (tensor<5x5x32x32xf32>) -> tensor<5x5x30x30xf32>
 //CHECK: %res_max_pool_single_out = onnx.MaxPoolSingleOut(%t26) {"onnx_node_name" = "/MaxPoolSingleOut", "auto_pad" = "VALID", "ceil_mode" = 0 : i64, "kernel_shape" = [3 : i64, 3 : i64], "dilations" = [1 : i64, 1 : i64], "pads" = [0 : i64, 0 : i64, 0 : i64, 0 : i64], "storage_order" = 0 : i64, "strides" = [1 : i64, 1 : i64]}: (tensor<5x5x32x32xf32>) -> tensor<5x5x30x30xf32>
+
+// CHECK-LABEL:  func.func @main_graph
```
Comment (compor) [inline tests/filecheck/dialects/onnx/onnx_ops.mlir:71]: is this check label needed here?

Comment (kayode-gif) [reply tests/filecheck/dialects/onnx/onnx_ops.mlir:71]: i didn't parse for me without it

Comment (superlopuh) [reply tests/filecheck/dialects/onnx/onnx_ops.mlir:71]: Hmm, that feels like an indication that something else is off. What's the error without the CHECK-LABEL?

**Review by superlopuh — APPROVED** [review]

File: tests/filecheck/dialects/onnx/onnx_ops.mlir (hunk 1 above)
Comment (compor) [reply tests/filecheck/dialects/onnx/onnx_ops.mlir:71]: I just removed it locally after checking out this PR branch and it seems to be working fine

**Pull request merged**